Common/cpp/SharedItems/MutableValue.cpp

```diff
@@ -87,18 +89,18 @@ std::vector<jsi::PropNameID> MutableValue::getPropertyNames(jsi::Runtime &rt) {
 }
 
 MutableValue::MutableValue(jsi::Runtime &rt, const jsi::Value &initial, NativeReanimatedModule *module):
-module(module), value(ShareableValue::adapt(rt, initial, module)) {}
+module(module), value(ShareableValue::adapt(rt, initial, module)) {
+}
 
-unsigned long int MutableValue::addListener(std::function<void ()> listener) {
-  unsigned long id = listeners.size() + 1;
-  listeners.push_back(std::make_pair(id, listener));
+unsigned long int MutableValue::addListener(unsigned long id, std::function<void ()> listener) {
+  listeners[id] = listener;
   return id;
 }
 
 void MutableValue::removeListener(unsigned long listenerId) {
-  listeners.erase(std::remove_if(listeners.begin(), listeners.end(), [=](const std::pair<unsigned long, std::function<void()>>& pair) {
-    return pair.first == listenerId;
-  }), listeners.end());
+  if (listeners.count(listenerId) > 0) {
```

Comment:
why do we need `count` here? Can't we just call `erase` regardless of whether the key exists or not?

Reply:
Looks like it's not necessary to check if the key is present in the map.
You are right It can be removed.